pkg/scale/encode_test.go

```diff
@@ -1127,8 +1134,16 @@ func Test_encodeState_encodeMap(t *testing.T) {
 			if err := es.marshal(tt.in); (err != nil) != tt.wantErr {
 				t.Errorf("encodeState.encodeMap() error = %v, wantErr %v", err, tt.wantErr)
 			}
-			if !reflect.DeepEqual(buffer.Bytes(), tt.want) {
-				t.Errorf("encodeState.encodeMap() = %v, want %v", buffer.Bytes(), tt.want)
+
+			gotExpectedEncode := false
+			for _, expectedValue := range tt.want {
+				if reflect.DeepEqual(buffer.Bytes(), expectedValue) {
+					gotExpectedEncode = true
+					break
+				}
+			}
+			if !gotExpectedEncode {
+				t.Errorf("encodeState.encodeMap() = %v, want %v", buffer.Bytes(), tt.want[0])
 			}
```

Comment:
nit you could use https://github.com/stretchr/testify/tree/master/assert with

```suggestion
			assert.Contains(t, tt.want, buffer.Bytes())
```